pkg/analysis/commentstart/testdata/src/a/a.go

```diff
@@ -34,3 +46,13 @@ type CommentStartTestStruct struct {
 
 // DoNothing is used to check that the analyser doesn't report on methods.
 func (CommentStartTestStruct) DoNothing() {}
+
```

Comment:
If you were to include a definition like below, do you think your logic would catch it? I suspect it may need a small tweak in checking the stack right? WDYT?
```
type {
  Foo struct {
    ...
  }
}

Reply:
It seems that `inspect.WithStack` will traverse all structs even if we have multiple declarations in one `type`, so it'll be checked.
Do you think we need a test for this?

Reply:
Yeah, could you anyway please, as in the future we may change the way the linters traverse to look only at objects accessible from the kubebuilder root annotation and I'd like to make sure we don't regress. Better to capture the case now while we are thinking about it than forget about it and regress in the future